.github/workflows/integration_test_juju3.yaml

```diff
@@ -0,0 +1,28 @@
+name: integration-tests-juju3
+
+on:
+  pull_request:
+  workflow_call:
+
+jobs:
+  integration-test-juju3:
+    name: Integration Test on Juju3
+    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
+    secrets: inherit
+    with:
+      extra-arguments: >-
+        --openstack-rc=${GITHUB_WORKSPACE}/openrc
+        --kube-config=${GITHUB_WORKSPACE}/kube-config
+        --screenshot-dir=/tmp
+      juju-channel: 3.1/stable
+      channel: 1.27-strict/stable
+      modules: '["test_addon", "test_core"]'
+      pre-run-script: |
+        -c "sudo microk8s enable hostpath-storage
+          sudo microk8s kubectl -n kube-system rollout status -w deployment/hostpath-provisioner
+          sudo microk8s config > ${GITHUB_WORKSPACE}/kube-config
+          chmod +x tests/integration/pre_run_script_juju3.sh
+          ./tests/integration/pre_run_script_juju3.sh
+          sed -i 's/juju>=2.9,<3/juju>=3.2.2/' ./tests/integration/requirements.txt"
```

Comment:
I think using `sed` to modify the version number in `requirements.txt` could be not very reliable due to potential changes in the Juju requirement declaration. I think it would be better to create a distinct `requirements-juju3.txt` specifically for Juju 3 and create a new tox environment, such as `integration-juju3` which would use the newly created `requirements-juju3.txt`. Additionally, the `test-tox-env` can be used to switch the workflow to use the new testing tox environment as needed.

Reply:
Sure, no worries.
I was aiming for a temporary solution here (assuming U Guys may fully switch to Juju3 soon).
However if Juju2 is to be supported in a long term, it's indeed better to have it reflected up to `tox` level